src/neorados/cls/log.cc

```
@@ -0,0 +1,142 @@
+  call.to_time = to;
+  call.marker = std::move(in_marker);
+  call.max_entries = max_entries;
+
+  encode(call, inbl);
+  op.exec("log", "list", inbl,
+          [entries, out_marker, truncated](error_code ec,
+                                           const buffer::list& bl) {
+            ::cls::log::ops::list_ret ret;
+            if (!ec) {
+              try {
+                auto iter = bl.cbegin();
+                decode(ret, iter);
+		if (entries)
+		  *entries = std::move(ret.entries);
+		if (out_marker)
+		  *out_marker = std::move(ret.marker);
+		if (truncated)
+		  *truncated = std::move(ret.truncated);
+              } catch (const buffer::error& err) {
+                // TODO: Currently there's no good way to propagate
+                // errors back.
+              }
+            }
+          });
+}
+
+void trim(neorados::WriteOp& op, ceph::real_time from_time,
+	  ceph::real_time to_time)
+{
+  bufferlist in;
+  ::cls::log::ops::trim_op call;
+  call.from_time = from_time;
+  call.to_time = to_time;
+  encode(call, in);
+  op.exec("log", "trim", in);
+}
+
+void trim(neorados::WriteOp& op, std::string from_marker,
+	  std::string to_marker)
+{
+  bufferlist in;
+  ::cls::log::ops::trim_op call;
+  call.from_marker = std::move(from_marker);
+  call.to_marker = std::move(to_marker);
+  encode(call, in);
+  op.exec("log", "trim", in);
+}
+
+void info(neorados::ReadOp& op, header* const h)
+{
+  buffer::list inbl;
+  ::cls::log::ops::info_op call;
+
+  encode(call, inbl);
+
+  op.exec("log", "info", inbl,
+          [h](error_code ec,
+              const buffer::list& bl) {
+            ::cls::log::ops::info_ret ret;
+            if (!ec) {
+              try {
+                auto iter = bl.cbegin();
+                decode(ret, iter);
+		if (h)
+                  *h = std::move(ret.header);
+              } catch (const buffer::error& err) {
+                // TODO: Currently there's no good way to propagate
+                // errors back.
```

where exactly does this get called? if we let the exception propagate, would the caller have a chance to catch it?